SteamBot/UserHandler.cs

```diff
@@ -190,6 +203,26 @@ public virtual void OnTradeRequestReply(bool accepted, string response)
 
         }
 
+        /// <summary>
+        /// Waits for the user to enter something into regular or trade chat, then returns it (as the result of a task)
+        /// Usage: The following displays "How many do you want to buy" and stores the user's response:
+        /// string userResponse = await GetUserResponse("How many do you want to buy?");
+        /// 
+        /// Note: calling this method causes the next user-message to NOT call OnMessage() or OnTradeMessage()
+        /// </summary>
+        /// <param name="message">An option message to display to the user.
+        /// Sent to whichever chat (normal or trade) is currently being used.</param>
+        protected virtual Task<string> GetUserResponse(string message = null)
+        {
+            if (message != null)
+            {
+                SendReplyMessage(message);
+            }
+
+            _waitingOnUserResponse = new TaskCompletionSource<string>();
```

Comment:
Huh, odd then cause I got compile errors when I merged it into MyBots branch of mine. CURSE YOU GIT MERGE!